src/components/ArticlesByTopic.jsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { fetchArticles } from "../Api";
+import { ListGroup, Button, Badge, Spinner } from "react-bootstrap";
+
+export default function ArticlesByTopic() {
+  const { topic } = useParams();
+  const [articles, setArticles] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    setIsLoading(true);
+    fetchArticles().then((articles) => {
+      setArticles(articles);
+      setIsLoading(false);
+    });
+  }, [topic]);
+
+  const filteredArticles = articles.filter(
+    (article) => article.topic === topic
```

Comment:
ultimately, you'll want to implement this filtering server side, because otherwise
- you're creating multiple sources of truth
- it's less efficient
- it's more expensive